### PR TITLE
fixed scheduler with variable LR + logging of LR + added save CSV files after each EPOCH for linear probing

### LINKED ISSUE #43: Fix the LinearProbingRunner omits self.lr_scheduler.step(), stopping the learning rate updates.

### PR DESCRIPTION
**Pull request by robertavram-md**
Fixes #43 

This pull request includes several updates to configuration files and code for training and testing models. The main changes involve updating file paths, modifying parameters, and adding new functionality for training with different configurations.

### Code Enhancements:

* [`projects/linear_probing_project.py`](diffhunk://#diff-307560652b389a7c4cb6cefeda189e49b9bcb2465a08bb8e23e62f8afb3267e1L113-L124): Updated optimizer initialization to use separate learning rates for backbone and heads.
* [`runners/linear_probing_runner.py`](diffhunk://#diff-92c255bf9d696b30762c71c9bb06108f16f6932d30859ab295813fec00696467R79-R94): Added functionality to check if the scheduler is per-iteration and updated the training loop to handle scheduler steps accordingly. Improved metrics accumulation and distributed gathering of predictions and targets. [[1]](diffhunk://#diff-92c255bf9d696b30762c71c9bb06108f16f6932d30859ab295813fec00696467R79-R94) [[2]](diffhunk://#diff-92c255bf9d696b30762c71c9bb06108f16f6932d30859ab295813fec00696467R164-R166) [[3]](diffhunk://#diff-92c255bf9d696b30762c71c9bb06108f16f6932d30859ab295813fec00696467R207-R237) [[4]](diffhunk://#diff-92c255bf9d696b30762c71c9bb06108f16f6932d30859ab295813fec00696467L227-R284)

### CODE REVIEW

File: runners/linear_probing_runner.py [hunk 1]
```diff
@@ -223,136 +247,77 @@ def _run_epoch(
...
+                        # Binary classification - compute AUC and AUPRC
+                        try:
+                            auc = roc_auc_score(targets_np, preds_np)
+                            auprc = average_precision_score(targets_np, preds_np)
+                            epoch_metrics[f"{mode}/{head}_auc"] = auc
+                            epoch_metrics[f"{mode}/{head}_auprc"] = auprc
+                        except ValueError as e:
+                            print(f"[DEBUG] rank={self.device} => Error computing metrics for {head}: {e}")
+                    else:
+                        # Multi-class classification - compute AUC for each class
```
Comment (JacquesDelfrate) [inline runners/linear_probing_runner.py:304]: pq avoir retire le confusion matrix ?

File: runners/linear_probing_runner.py [hunk 2]
```diff
@@ -550,30 +547,61 @@ def _save_predictions(
...
             df: pd.DataFrame = pd.DataFrame(predictions_dict)
             predictions_dir: str = os.path.join(self.output_dir, "predictions") 
             os.makedirs(predictions_dir, exist_ok=True)
-            output_file: str = os.path.join(
+            
+            # Save current epoch predictions
+            current_epoch_file: str = os.path.join(
                 predictions_dir, 
-                f'{mode}_predictions_epoch_{epoch}.csv'
+                f'val_predictions_epoch_{epoch}.csv'
```
Comment (JacquesDelfrate) [inline runners/linear_probing_runner.py:583]: svp laisser {mode}_predictions_epoch_{epoch}.csv
On pourrait vouloir appeler cette fonction pour d'autre mode que val comme inference.

File: runners/linear_probing_runner.py [hunk 3]
```diff
@@ -550,30 +547,61 @@ def _save_predictions(
...
-            df.to_csv(output_file, index=False)
-            print(f"[DEBUG] rank={self.device} => Saved predictions to {output_file}")
+            df.to_csv(current_epoch_file, index=False)
+            print(f"[DEBUG] rank={self.device} => Saved epoch {epoch} predictions to {current_epoch_file}")
+            
+            # If this is the best epoch, also save as best predictions
+            if epoch == self.best_epoch:
+                best_file: str = os.path.join(
+                    predictions_dir, 
+                    f'val_predictions_best.csv'
```
Comment (JacquesDelfrate) [inline runners/linear_probing_runner.py:592]: pareil ici utiliser {mode}_predictions_best_epoch_{epoch}.csv

File: runners/linear_probing_runner.py [hunk 4]
```diff
@@ -145,6 +160,9 @@ def train(
                 device_ids=self.device
             )
             
+            if self.lr_scheduler and (not self.scheduler_per_iteration):
```
[the remark this answers is not in the document]
Comment (robertavram-md) [reply runners/linear_probing_runner.py:163]: No stress

File: runners/linear_probing_runner.py (hunk 1 above)
Comment (robertavram-md) [reply runners/linear_probing_runner.py:304]: Vibe coded

File: runners/linear_probing_runner.py (hunk 2 above)
Comment (robertavram-md) [reply runners/linear_probing_runner.py:583]: Done

**Comment by robertavram-md** [comment]
@JacquesDelfrate Ready for your review + merge
